### PR TITLE
[FLINK-3678] Make Flink logs directory configurable

### PR DESCRIPTION
**Pull request by stefanobaghino**
* Edit config.sh
* Document the newly defined log directory configuration key

### CODE REVIEW

**Comment by chiwanpark** [comment]
I tested this PR in local cluster. Looks good to me. +1

**Comment by StephanEwen** [comment]
Looks good to me!

File: docs/setup/config.md [hunk 1]
```diff
@@ -275,6 +275,10 @@ For example when running Flink on YARN on an environment with a restrictive fire
 
 - `recovery.job.delay`: (Default 'akka.ask.timeout') Defines the delay before persisted jobs are recovered in case of a recovery situation. 
 
+## Environment
+
+- `env.log.dir`: (Defaults to the `log` directory under Flink's home) Defines the directory where the Flink logs are saved.
```
Comment (uce) [inline docs/setup/config.md]: Are relative paths resolved relative to Flink's root dir?

Comment (stefanobaghino) [inline docs/setup/config.md]: No, the user has to input an absolute path. I wouldn't recommend adding this option, a possible improvement would be to document this restriction and to check at startup for malformed paths. Would `config.sh` be a good place to perform this check?

Comment (uce) [inline docs/setup/config.md]: Yes, adding a comment would be very good! A check even better. :+1:

Comment (stefanobaghino) [inline docs/setup/config.md]: I did some research but checking for a valid path across all possible operative systems and filesystems would be prohibitive, I guess we'll have to settle for the documentation at this time. :smiley: Thanks for pointing it out!

**Comment by stefanobaghino** [comment]
I think this is ready.

**Comment by uce** [comment]
Yes, I'm going to merge it later today.

**Comment by stefanobaghino** [comment]
Thanks! Hope it will make it to 1.0.1. :smile:

**Comment by rmetzger** [comment]
I'll merge it into 1.0.3

**Pull request closed without merging**